packages/codegen-ui-react/lib/amplify-ui-renderers/form.ts

```diff
@@ -103,15 +108,10 @@ export default class FormRenderer extends ReactComponentRenderer<BaseComponentPr
                 ),
               ),
               factory.createExpressionStatement(
-                /**
-                 * TODO: pass in props from functional argument
-                 * for datastore it will be onSubmitBefore & onSubmitComplete
-                 * for byod it will only be onSubmit override function
-                 */
                 factory.createCallExpression(
-                  factory.createIdentifier(getActionIdentifier(this.form.name, 'onSubmit')),
+                  factory.createIdentifier(getActionIdentifier(name, 'onSubmit')),
                   undefined,
-                  [],
+                  dataSourceType === 'DataStore' ? [] : [factory.createIdentifier(getFormFieldStateName(name))],
```

Comment:
If using a custom form we want to use the passed into function.
We'll also want to pass in the fields into this said function

```jsx
<form
      onSubmit={async (event) => {
        event.preventDefault();
        customDataFormOnSubmit(customDataFormFields);
      }}
```